src/jit/compiler.h

```diff
@@ -3005,6 +3005,10 @@ class Compiler
         bool CanPromoteStructType(CORINFO_CLASS_HANDLE typeHnd);
         bool TryPromoteStructVar(unsigned lclNum);
 
+#ifdef DEBUG
+        void CheckFakedType(CORINFO_FIELD_HANDLE fieldHnd, var_types requestedType);
```

Comment:
Would be nice to find an alternative to "faked". Can't say I have a better idea though. Maybe "scalar struct type" - a struct containing a single scalar type field?

Reply:
What about `Retyped(As)ScalarFieldsMap`?

Reply:
Yep, sounds fine